cfme/utils/appliance/__init__.py

```diff
@@ -453,7 +453,12 @@ def server(self):
     def _rest_api_server(self):
         shref = self.appliance.rest_api.server_info['server_href']
         results = self.appliance.rest_api.collections.servers.all
-        server, = (r for r in results if r.href == shref)
+        try:
+            server, = (r for r in results if r.href == shref)
+        except ValueError:
```

Comment:
This error is sometimes encountered while dealing with customer database.

Reply:
@valaparthvi Can you use

```python,
return self.server.rest_api_entity
```

Here instead? It includes the `ValueError` already. https://github.com/ManageIQ/integration_tests/blob/07175beba8c27d731a1e62d626dd6f4dd495aa84/cfme/base/__init__.py#L110-L115

Also can this `_rest_api_server` be a property?

Reply:
I don't think placing the ValueError code inside `rest_api_entity` would work. Afaik this is how the chain is:
`appliance.server` uses `_rest_api_server` and you will use `rest_api_entity` like `appliance.server.rest_api_entity`. We are getting the error in `_rest_api_server` so it doesn't make sense to move it to `rest_api_entity`. But your suggestion about making it property is good, I'll work on that.

Reply:
When is this hitting this condition - what are you trying to solve here?

This was explicitly raising the valueerror, because there is an issue here - the server href returned by the API is not actually found in the servers list. There should be one, and only one, server matching this. I don't think its good behavior to just fallback to the master server for this.

Reply:
I am hitting this condition when I restore the CU DB on the appliance.
The problem is fetching `appliance.rest_api.collections.servers.all` returns all the servers restored by the CU  DB, but `appliance.rest_api.server_info["server_href"]` still returns the old the server href. It looks like a bug, I'll file a bug, but until then can we keep it as a workaround? Or it doesn't look like a good idea?

Reply:
So, I found that deleting the `rest_api` cached property after restoring the database did the trick.